bundles/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/util/FsUtil.java

```diff
@@ -0,0 +1,308 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.ui.cometvisu.internal.util;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Random;
+import java.util.regex.Pattern;
+import java.util.zip.CRC32;
+import java.util.zip.CheckedInputStream;
+import java.util.zip.CheckedOutputStream;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.openhab.ui.cometvisu.internal.ManagerSettings;
+import org.openhab.ui.cometvisu.internal.MountPoint;
+import org.openhab.ui.cometvisu.internal.backend.model.rest.FsEntry;
+import org.openhab.ui.cometvisu.internal.backend.model.rest.FsEntry.TypeEnum;
+import org.openhab.ui.cometvisu.internal.backend.model.rest.ReadResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Filesystem helper that supports all file operation in the installed CometVisus config folder
+ *
+ * @author Tobias Bräutigam - initial contribution
+ *
+ */
+public class FsUtil {
```

Comment:
It is very difficult to understand what this class is capable of. Is the JavaDoc correct that all operations are restricted to the cometvisu config folder, which resides somewhere in openHAB's userdata folder? Would be great, if that would be guaranteed.

Reply:
Yes thats right, if I did not make a huge mistake all file operations are restricted to that folder